core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -99,6 +102,68 @@ case object ProcessTreeMetrics extends ExecutorMetricType {
   }
 }
 
+case object GarbageCollectionMetrics extends ExecutorMetricType with Logging {
+  import GC_TYPE._
+  override val names = Seq(
+    "MinorGCCount",
+    "MinorGCTime",
+    "MajorGCCount",
+    "MajorGCTime"
+  )
+
+  private lazy val supported: Boolean = {
+    val shouldLogStageExecutorGCMetrics =
+      SparkEnv.get.conf.get(config.EVENT_LOG_PROCESS_TREE_METRICS)
+    val supportedVendor = System.getProperty("java.vendor").contains("Oracle") ||
+      System.getProperty("java.vendor").contains("OpenJDK")
+    shouldLogStageExecutorGCMetrics && supportedVendor
```

Comment:
While `getGarbageCollectorMXBeans()` is a part of stardard API, why does this PR allow only two JVMs vendor to enable this?

Reply:
I thought it's version specific as well and I am only sure those venders are safe. I will remove this since this API are standard in Java Lang.

Reply:
Thanks for your review. Also I disabled this feature by default.